include/xsimd/arch/generic/xsimd_generic_memory.hpp

```diff
@@ -55,6 +55,42 @@ namespace xsimd
             return batch<T, A>::load_aligned(concat_buffer);
         }
 
+        // gather
+        namespace detail
+        {
+            template <typename T, typename A, typename U, typename V, size_t I>
+            inline batch<T, A> gather(U const* src, batch<V, A> const& index,
+                                      ::xsimd::detail::index_sequence<I>) noexcept
+            {
+                return insert(batch<T, A> {}, static_cast<T>(src[index.get(I)]),
+                              ::xsimd::index<I>());
+            }
+
+            template <typename T, typename A, typename U, typename V, size_t I,
+                      size_t... Is,
+                      typename = typename std::enable_if<sizeof...(Is) != 0>::type>
+            inline batch<T, A>
+            gather(U const* src, batch<V, A> const& index,
+                   ::xsimd::detail::index_sequence<I, Is...>) noexcept
+            {
+                const auto test = kernel::detail::gather<T, A, U, V>(
+                    src, index, ::xsimd::detail::index_sequence<Is...>());
+                return insert(test, static_cast<T>(src[index.get(I)]),
```

Comment:
interesting approach to use insert here. Unfortunately for some arch we don't have a specialized version of insert and I fear it would be slower than the naive *fill a temporary buffer then aligned load* approach. Probably needs benchmarking.

Reply:
This Godbolt compares the codegen of the two: https://godbolt.org/z/GM7cYeahe

And this one uses the two alternatives in context (deinterleaving of packed pixels): https://godbolt.org/z/eKGhEqWYq

Reply:
proof by benchmark, perfect :-)